### PR TITLE
Get pfp

### PR DESCRIPTION
**Pull request by cccyyyr**
Load-pfp outputs a stream.

### CODE REVIEW

**Comment by crchong1** [comment]
@cccyyyr what is the status of this PR?

File: src/main/Config/Message.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package Config;
+
+import org.json.JSONObject;
+
```
Comment (kofmangregory) [inline src/main/Config/Message.java:4]: Can you add a comment describing what this class does?

File: src/main/Config/Service.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package Config;
+
```
Comment (kofmangregory) [inline src/main/Config/Service.java:2]: Add a comment to explain what this class is for.

File: src/main/ForFun/Sub.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package ForFun;
```
Comment (kofmangregory) [inline src/main/ForFun/Sub.java:1]: Should this file be deleted?

File: src/main/ForFun/Super.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
+package ForFun;
```
Comment (kofmangregory) [inline src/main/ForFun/Super.java:1]: Should this file be deleted?

File: src/test/OrgTests/OrganizationJWTTests.java [hunk 5]
```diff
@@ -196,102 +196,102 @@ public void invalidUserType() {
...
-    assert (resp.has("message"));
-    assertThat(resp.getString("message")).isEqualTo("Success.");
-    assert (resp.has("status"));
-    assertThat(resp.getString("status")).isEqualTo("SUCCESS");
-    assert (resp.has("workers"));
-    // assertThat(resp.getInt("workers")).isEqualTo(12);
-    assert (resp.has("directors"));
-    // assertThat(resp.getInt("directors")).isEqualTo(2);
-  }
+  //  @Test
```
Comment (kofmangregory) [inline src/test/OrgTests/OrganizationJWTTests.java:199]: Can we delete all commented out code? Here and elsewhere in the PR.

File: src/main/User/UserController.java [hunk 6]
```diff
@@ -45,6 +54,49 @@ public UserController(MongoDatabase db) {
     logger = l.createLogger("UserController");
   }
 
+  public InputStream getUserPfp(String username) {
+    Bson filter = Filters.eq("metadata.owner", username);
+    GridFSBucket gridBucket = GridFSBuckets.create(db, "pfp");
+    GridFSFile grid_out = gridBucket.find(filter).first();
```
Comment (kofmangregory) [inline src/main/User/UserController.java:60]: camelCase

File: src/main/User/UserController.java [hunk 7]
```diff
@@ -45,6 +54,49 @@ public UserController(MongoDatabase db) {
     logger = l.createLogger("UserController");
   }
 
+  public InputStream getUserPfp(String username) {
+    Bson filter = Filters.eq("metadata.owner", username);
+    GridFSBucket gridBucket = GridFSBuckets.create(db, "pfp");
```
Comment (kofmangregory) [inline src/main/User/UserController.java:59]: Can you add some comments about what these methods do?

File: src/main/User/UserController.java [hunk 8]
```diff
@@ -45,6 +54,49 @@ public UserController(MongoDatabase db) {
...
+    return pfp;
+  }
+
+  public void uploadPfp(UploadedFile file, String username) {
+    String fileName = file.getFilename();
+    InputStream content = file.getContent();
+    GridFSBucket gridBucket = GridFSBuckets.create(db, "pfp");
+    GridFSUploadOptions options =
+        new GridFSUploadOptions()
+            .chunkSizeBytes(100000)
```
Comment (kofmangregory) [inline src/main/User/UserController.java:74]: Can this constant be declared at the top of the file?

File: src/main/User/UserController.java [hunk 9]
```diff
@@ -45,6 +54,49 @@ public UserController(MongoDatabase db) {
...
+    GridFSBucket gridBucket = GridFSBuckets.create(db, "pfp");
+    GridFSFile grid_out = gridBucket.find(filter).first();
+    if (grid_out == null || grid_out.getMetadata() == null) {
+      return null;
+    }
+    InputStream pfp = gridBucket.openDownloadStream(grid_out.getObjectId());
+    return pfp;
+  }
+
+  public void uploadPfp(UploadedFile file, String username) {
```
Comment (kofmangregory) [inline src/main/User/UserController.java:68]: Are there any restrictions on the size of `file`?

File: src/main/User/UserController.java [hunk 10]
```diff
@@ -45,6 +54,49 @@ public UserController(MongoDatabase db) {
...
+        uploadPfp(file, username);
+        logger.info(username + " has successfully uploaded a profile picture");
+        ctx.json(UserMessage.SUCCESS.toJSON("Profile Picture Uploaded Successfully").toString());
+      };
+
+  public Handler loadPfp =
+      ctx -> {
+        String username = ctx.sessionAttribute("username");
+        InputStream pfp = getUserPfp(username);
+        ctx.header("Content-Type", "application/pfp");
```
Comment (kofmangregory) [inline src/main/User/UserController.java:96]: Is this a valid content-type?

File: src/main/User/UserController.java [hunk 11]
```diff
@@ -45,6 +54,49 @@ public UserController(MongoDatabase db) {
...
+                    .append("owner", username));
+    gridBucket.uploadFromStream(fileName, content, options);
+  }
+
+  public Handler uploadPfp =
+      ctx -> {
+        String username = ctx.sessionAttribute("username");
+        logger.info(username + " is Attempting to upload a profile picture");
+        UploadedFile file = ctx.uploadedFile("file");
+        uploadPfp(file, username);
```
Comment (kofmangregory) [inline src/main/User/UserController.java:87]: Does this function ever throw an error? Is it possible to unsuccessfully upload a profile pic? Same comment below for loading one.

File: src/test/UserTest/PfpTest.java [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+    TestUtils.startServer();
+    TestUtils.setUpTestDB();
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    TestUtils.tearDownTestDB();
+  }
+
+  @Test
```
Comment (kofmangregory) [inline src/test/UserTest/PfpTest.java:38]: Can we place tests below all functions declared in this file?

File: src/test/UserTest/PfpTest.java [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+            .header("Content-Disposition", "attachment")
+            .field("file", examplePDF)
+            .asString();
+    JSONObject uploadResponseJSON = TestUtils.responseStringToJSON(uploadResponse.getBody());
+    assertThat(uploadResponseJSON.getString("status")).isEqualTo("SUCCESS");
+  }
+
+  public static void getPfp() {
+    HttpResponse get = Unirest.post(TestUtils.getServerUrl() + "/load-pfp").asBytes();
+    assert (get.isSuccess());
```
Comment (kofmangregory) [inline src/test/UserTest/PfpTest.java:59]: Can we ensure that the pic we got back is correct?

**Review by kofmangregory — CHANGES_REQUESTED** [review]

**Comment by crchong1** [comment]
Wait if there is a new pfp PR, what should we do about this PR? @cccyyyr

**Pull request closed without merging**